server/utils/in-memory-cache.js

```diff
@@ -1,6 +1,7 @@
 const localCache = new Map()
 
 localCache.exists = key => localCache.has(key)
+localCache.expire = (key, ttl) => setTimeout(() => localCache.delete(key), ttl * 1000)
```

Comment:
General shout-out that this is the kind of situation where Typescript is really valuable. If you were using Typescript, the missing `expire` function would have been caught as a compiler error rather than during code review.

Just food for thought, not something I expect you to change in this PR.